MediaGalleryUi/view/adminhtml/ui_component/media_gallery_listing.xml

```diff
@@ -94,6 +94,7 @@
         <column name="overlay" component="Magento_Ui/js/grid/columns/overlay">
             <settings>
                 <label translate="true">Overlay</label>
+                <bodyTmpl>Magento_MediaGalleryUi/grid/columns/overlay</bodyTmpl>
```

Comment:
It's possible to use `overlay` key in the grid data to output the required overlay without the need to override body template:
 - `overlay: "Unlicensed"` results into "Unlicensed" overlay
- `overlay: false` (or just missing the overlay key) results into no overlay